modules/renter/contractor/update.go

```diff
@@ -53,16 +53,16 @@ func (c *Contractor) ProcessConsensusChange(cc modules.ConsensusChange) {
 			}
 			defer c.editLock.Unlock()
 
-			// renew any contracts that have entered the renew window
+			// renew any (online) contracts that have entered the renew window
 			err := c.managedRenewContracts()
 			if err != nil {
 				c.log.Debugln("WARN: failed to renew contracts after processing a consensus chage:", err)
 			}
 
-			// if we don't have enough contracts, form new ones
+			// if we don't have enough (online) contracts, form new ones
 			c.mu.RLock()
 			a := c.allowance
-			remaining := int(a.Hosts) - len(c.contracts)
+			remaining := int(a.Hosts) - len(c.onlineContracts())
```

Comment:
oh hmm.

I guess that's sufficient if you trust the race detector. You can't trigger a deadlock this way, only a race condition.

At some point in my head I switched it from not doing locking to requiring a write lock.